### PR TITLE
Current Location Bottom Border

### PR DESCRIPTION
**Pull request by mwilkins74**
Added bottom border to whatever the current location is via this prop:
`isSelected={location.pathname === enumMappings[linkEnum].url}`

Screen Shot 2024-03-06 at 4 33 14 PM

### CODE REVIEW

**Comment by mwilkins74** [comment]
Ok, got npm install running so back up and running. 
And, for some reason, you are right, the work I did to add in the bottom border must not have saved so I pushed an "empty" PR for all intents and purposes. 
I'll redo it tomorrow.

**Comment by mwilkins74** [comment]
Real professional of me *facepalm*

**Comment by ChuckBTaylor** [comment]
Is this MR ready for review?

**Comment by mwilkins74** [comment]
Yes

File: src/components/molecules/NavList.tsx [hunk 1]
```diff
@@ -13,24 +14,36 @@ const orderedLinks: (Genre | Role)[] = [
...
           label: enumMappings[linkEnum].label,
         }
-        return <NavLinks navLink={navLink} key={`nav-link-${linkEnum}`} />
+        return (
+          <NavLinks
+            navLink={navLink}
+            key={`nav-link-${linkEnum}`}
+            isSelected={location.pathname === enumMappings[linkEnum].url}
+          />
+        )
```
Comment (ChuckBTaylor) [inline src/components/molecules/NavList.tsx:32]: :art: I see this and lines 40-44 and see duplicate code. Is there a way to refactor this to skip the duplication?

That being said, it's not a lot and creating a function for this alone might be overkill.  6-of-1

File: src/components/molecules/navLinks/NavLinks.tsx [hunk 2]
```diff
@@ -3,14 +3,20 @@ import { Link } from 'react-router-dom'
 
 type NavLinksProps = {
   navLink: { route: string; label: string }
+  isSelected?: boolean
```
Comment (ChuckBTaylor) [inline src/components/molecules/navLinks/NavLinks.tsx:6]: ❓ Is this really optional?

File: src/components/molecules/navLinks/NavLinks.tsx [hunk 3]
```diff
@@ -3,14 +3,20 @@ import { Link } from 'react-router-dom'
...
 }
 
-export function NavLinks({ navLink }: NavLinksProps) {
+export function NavLinks({ navLink, isSelected }: NavLinksProps) {
```
Comment (ChuckBTaylor) [inline src/components/molecules/navLinks/NavLinks.tsx:9]: 🔧 If `isSelected` is optional, I'd prefer not to leave it `undefined`. In general, nothing should be `undefined` Can we set it to a value here?

```suggestion
export function NavLinks({ navLink, isSelected = false }: NavLinksProps) {
```

**Review by ChuckBTaylor — APPROVED** [review]

File: src/components/molecules/NavList.tsx (hunk 1 above)
Comment (mwilkins74) [reply src/components/molecules/NavList.tsx:32]: I rewrote the `secondNavLinks` and `firstNavLinks` like this utilizing a new function called `generateNavLinks`. 
I saved the re-written code in a slack message so if you like this idea better I can easily come back in and change this file. 

```
import { List } from '@mui/material'
import { useMemo } from 'react'
import { enumMappings } from '../../common/constants/constants'
import { Genre, Role, navLinks } from '../../common/constants/enums'
import { NavLinks } from './navLinks/NavLinks'
import { useLocation } from 'react-router-dom'

const orderedLinks: (Genre | Role)[] = [
  'animation',
  'documentary',
  'director_of_photography',
  'director_producer',
  'series',
]

export function NavList() {
  const location = useLocation()

  const generateNavLink = (navLink: { route: string; label: string }, isSelected: boolean, key: string) => (
    <NavLinks
      navLink={navLink}
      key={key}
      isSelected={isSelected}
    />
  )

  const secondNavLinks = useMemo(
    () =>
      orderedLinks.map(linkEnum => {
        const navLink = {
          route: enumMappings[linkEnum].url,
          label: enumMappings[linkEnum].label,
        }
        return generateNavLink(navLink, location.pathname === enumMappings[linkEnum].url, `nav-link-${linkEnum}`)
      }),
    [location.pathname]
  )

  const firstNavLinks = useMemo(
    () =>
      navLinks.map((item, index) => (
        generateNavLink(item, location.pathname === item.route, `nav-link-${item.label}-${index}`)
      )),
    [location.pathname]
  )

  return (
    <div
      style={{
        height: '100%',
        backgroundColor: 'black',
        fontWeight: 'bold',
        color: 'white',
        fontSize: 20,
        textAlign: 'right',
        overflowY: 'hidden',
        paddingLeft: 10,
        paddingRight: 10,
      }}
    >
      <List
        sx={{
          width: '100%',
          paddingBottom: '20px',
          maxWidth: 360,
        }}
        component="nav"
      >
        {firstNavLinks}
      </List>

      <List
        sx={{
          width: '100%',
          height: '100%',
          maxWidth: 360,
        }}
        component="nav"
      >
        {secondNavLinks}
      </List>
    </div>
  )
}
```

File: src/components/molecules/navLinks/NavLinks.tsx (hunk 3 above)
Comment (mwilkins74) [reply src/components/molecules/navLinks/NavLinks.tsx:9]: I removed the option as the more I thought about it `isSelected` is needed for the styling to trigger as we desire.

**Pull request merged**